microcosm/caching.py

```diff
@@ -7,36 +7,20 @@
 (or to force re-creation).
 
 """
-from abc import ABCMeta, abstractmethod
 from collections import defaultdict
-from typing import Mapping, Any
+from collections.abc import MutableMapping
+from typing import Any, Mapping
 
 
-class Cache(metaclass=ABCMeta):
+class Cache(MutableMapping):
     """
-    A cache uses the basic dictionary interface and defines a name.
+    A cache supports the basic dictionary interface and defines a name.
 
     """
     @classmethod
     def name(self):
         raise NotImplementedError
 
-    @abstractmethod
-    def __contains__(self, name):
-        pass
-
-    @abstractmethod
-    def __getitem__(self, name):
-        pass
-
-    @abstractmethod
-    def __setitem__(self, name, component):
-        pass
-
-    @abstractmethod
-    def __delitem__(self, name):
-        pass
-
```

Comment:
These are all defined by `collections.abc.MutableMapping`

Reply:
I'm guessing the definition in the base class will work for our `Cache` class?

Reply:
Yeah these are mostly all abstract in the base class and the ones that are not function as expected